File: algebra/geometric_progression/javascript/geometric_progression.js
// For a geometric progression a + ar + ar^2 + ar^3 + ...
// with a being the scale factor and r being the common ratio

// This function calculates the nth term
function nth_term(a, r, n){
	return  a * Math.pow(r, n - 1);
}

// This function calculates the sum of the first n terms
function sum_of_first_n(a, r, n){
	return (a * (1-Math.pow(r,n))) / (1 - r);
}

File: algebra/geometric_progression/java/geometric_progression.java
// For a geometric progression a + ar + ar^2 + ar^3 + ...
// with a being the scale factor and r being the common ratio


public class geometric_progression {

    // This function calculates the nth term.
    public static int nth_term(int a, int r, int n) {
        return a * ((int) Math.pow(r, n - 1));
    }

    //This function calculates sum of first n terms 
    public static int sum_of_first_n_tems(int a, int r, int n) {
        if (r == 1) {
            return a * n;
        } else {
            return a * ((int) Math.pow(r, n) - 1) / (r - 1);
        }
    }

}
